### DISCUSSION TITLE
Non-standard query in redirect URL when using email auth

### DISCUSSION THREAD

**Discussion by kaylendog** [supabase/supabase#14260] [Questions] [answered]
Hello! I'm using Supabase's email auth with confirm enabled, but I am getting a non-standard query in the returned redirect URL.

I'm manually setting the email redirect for now while in development - I have my site URL configured to the production endpoint but have added `http://localhost:3000/**` as a redirect URL.
```ts
const { data, error } = await this.auth.signUp({
    email,
    password,
    options: {
        emailRedirectTo: "http://localhost:3000/api/v1/auth/verify",
    },
});
```
The confirm link in the email is as I would expect it
```
https://<SUBDOMAIN>.supabase.co/auth/v1/verify?token=<TOKEN>&type=signup&redirect_to=http://localhost:3000/api/v1/auth/verify
```
But on redirect the URL is expanded into
```
http://localhost:3000/api/v1/auth/verify#access_token=<ACCESS_TOKEN>&expires_in=3600&refresh_token=<REFRESH_TOKEN>&token_type=bearer&type=signup
```
The query is indicated by `#` which is very non-standard. Is this a configuration issue on my end, or do I need to parse this URL separately?

**Comment by GaryAustin1** [comment]
The 2nd link seems to be the URL supabase uses to call your app with the validated token.  The hash is the standard way they send it.
Your first link is usually the email link that goes back to Supabase to confirm your user's email and then do the call to the next link.

**Reply by kaylendog** [reply]
Any idea why the hash was chosen over the standard `?` char? Makes handling it more difficult...

**Reply by GaryAustin1** [reply, marked as answer]
Many issues on this in supabase/gotrue over the past two years.
`This was actually intended as a security feature since ? params can be read by the server (and thus be logged together with access token) whereas # params remain in the browser. ` mentioned in this link: https://github.com/supabase/gotrue/issues/1011 as one example.

The link also mentions with the new pkce flow they may be able to change this.

**Reply by kaylendog** [reply]
Ah I see, thank you!
